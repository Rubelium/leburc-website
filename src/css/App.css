.app {
  text-align: center;
  background-color: #00AEAE;
}

.app-title {
  height: 5vmin;
  pointer-events: none;
}

.app-logo {
  height: 30vmin;
  pointer-events: none;
}

.app-link {
  color: #000000;
  text-decoration: none;
}

.app-header {
  min-height: 60vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.app-body {
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #FFFFFF;
}

.app-footer {
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #212529;
  color: #FFFFFF;
}

.media-player-wrapper {
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.basic-navbar-nav {
  font-size: 20px;
}

@media (prefers-reduced-motion: no-preference) {
  .app-logo {
    animation: app-logo-spin infinite 40s linear;
  }
}

@keyframes app-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}