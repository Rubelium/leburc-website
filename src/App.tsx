import logo from "./png/logoF2t4BlackWhiteCircleFilled.png";
import "./css/App.css";

const App = () => {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          FLOOR<sup>4</sup>
        </p>
        <a
          className="App-link"
          href="https://soundcloud.com/f2t4"
          target="_blank"
          rel="noopener noreferrer"
        >
          SoundCloud
        </a>
        <a
          className="App-link"
          href="https://www.youtube.com/F2t4TV"
          target="_blank"
          rel="noopener noreferrer"
        >
          YouTube
        </a>
        <a
          className="App-link"
          href="https://www.instagram.com/f2t4/"
          target="_blank"
          rel="noopener noreferrer"
        >
          Instagram
        </a>
        <a
          className="App-link"
          href="https://twitter.com/_F2t4"
          target="_blank"
          rel="noopener noreferrer"
        >
          Twitter
        </a>
      </header>
    </div>
  );
};

export default App;
