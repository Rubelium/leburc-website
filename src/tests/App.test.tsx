import { render, screen } from "@testing-library/react";
import App from "../App";

describe("it should render", () => {
  test("the snapshot of the <App /> component", () => {
    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });

  test("the social media links", () => {
    render(<App />);

    const soundCloudLinkElement = screen.getByText(/SoundCloud/i);
    const youTubeLinkElement = screen.getByText(/YouTube/i);
    const InstagramLinkElement = screen.getByText(/Instagram/i);
    const TwitterLinkElement = screen.getByText(/Twitter/i);

    expect(soundCloudLinkElement).toBeInTheDocument();
    expect(youTubeLinkElement).toBeInTheDocument();
    expect(InstagramLinkElement).toBeInTheDocument();
    expect(TwitterLinkElement).toBeInTheDocument();
  });
});
