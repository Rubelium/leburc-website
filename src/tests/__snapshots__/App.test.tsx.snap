// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the snapshot of the <App /> component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logoF2t4BlackWhiteCircleFilled.png"
          />
          <p>
            FLOOR
            <sup>
              4
            </sup>
          </p>
          <a
            class="App-link"
            href="https://soundcloud.com/f2t4"
            rel="noopener noreferrer"
            target="_blank"
          >
            SoundCloud
          </a>
          <a
            class="App-link"
            href="https://www.youtube.com/F2t4TV"
            rel="noopener noreferrer"
            target="_blank"
          >
            YouTube
          </a>
          <a
            class="App-link"
            href="https://www.instagram.com/f2t4/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Instagram
          </a>
          <a
            class="App-link"
            href="https://twitter.com/_F2t4"
            rel="noopener noreferrer"
            target="_blank"
          >
            Twitter
          </a>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logoF2t4BlackWhiteCircleFilled.png"
        />
        <p>
          FLOOR
          <sup>
            4
          </sup>
        </p>
        <a
          class="App-link"
          href="https://soundcloud.com/f2t4"
          rel="noopener noreferrer"
          target="_blank"
        >
          SoundCloud
        </a>
        <a
          class="App-link"
          href="https://www.youtube.com/F2t4TV"
          rel="noopener noreferrer"
          target="_blank"
        >
          YouTube
        </a>
        <a
          class="App-link"
          href="https://www.instagram.com/f2t4/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Instagram
        </a>
        <a
          class="App-link"
          href="https://twitter.com/_F2t4"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </a>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
